lhs.js

```diff
@@ -116,14 +148,17 @@ class DomLhs extends Lhs {
         this.selector = selector;
     }
 
+    clone() {
+        return new this.constructor(this.selector);
+    }
+
     fnodes(ruleset) {
-        const matches = ruleset.doc.querySelectorAll(this.selector);
         const ret = [];
-        for (let i = 0; i < matches.length; i++) {  // matches is a NodeList, which doesn't conform to iterator protocol
-            const element = matches[i];
-            ret.push(ruleset.fnodeForElement(element));
+        const matches = ruleset.doc.querySelectorAll(this.selector);
+        for (let i = 0; i < matches.length; i++) {
```

Comment:
You could wrap `matches` in a call to `Array.from`, so you can use the iterator protocol:

```js
fnodes(ruleset) {
    const matches = ruleset.doc.querySelectorAll(this.selector);
    const ret = Array.from(matches).map(match => ruleset.fnodeForElement(match));
    return super.fnodesSatisfyingWhen(ret);
}

Reply:
I did originally have it using Array.from, but Erik preferred the for loop!

Reply:
(I was trying to avoid copying the whole thing.)